Apps/Admin/Client/Pages/PatientDetailsPage.razor.cs

```diff
@@ -0,0 +1,115 @@
+
+        [Inject]
+        private IState<PatientSupportState> PatientSupportState { get; set; } = default!;
+
+        [Inject]
+        private NavigationManager NavigationManager { get; set; } = default!;
+
+        private bool MessagingVerificationsLoading => this.MessageVerificationState.Value.IsLoading;
+
+        private bool HasMessagingVerificationsError => this.MessageVerificationState.Value.Error is { Message.Length: > 0 };
+
+        private IEnumerable<MessagingVerificationModel> MessagingVerifications => this.MessageVerificationState.Value.Data ?? Enumerable.Empty<MessagingVerificationModel>();
+
+        private bool PatientsLoaded => this.PatientSupportState.Value.Loaded;
+
+        private bool HasPatientsError => this.PatientSupportState.Value.Error is { Message.Length: > 0 };
+
+        private bool HasPatientsWarning => this.PatientSupportState.Value.WarningMessages.Any();
+
+        private PatientSupportDetails? Patient =>
+            this.PatientSupportState.Value.Result?.SingleOrDefault(x => x.Hdid == this.Hdid);
+
+        private string PatientName => StringManipulator.JoinWithoutBlanks(new[] { this.Patient?.PreferredName?.GivenName, this.Patient?.PreferredName?.Surname });
+
+        private string? Warning => this.Patient == null ? null : MapStatusToWarning(this.Patient.Status);
+
+        private string? Hdid { get; set; }
+
+        /// <inheritdoc/>
+        protected override void OnInitialized()
+        {
+            base.OnInitialized();
+
+            Uri uri = this.NavigationManager.ToAbsoluteUri(this.NavigationManager.Uri);
+            if (QueryHelpers.ParseQuery(uri.Query).TryGetValue("hdid", out StringValues hdid))
+            {
+                this.Hdid = hdid;
+                this.RetrievePatientData();
+            }
+            else
+            {
+                this.NavigationManager.NavigateTo("/support");
+            }
+        }
+
+        private static string? MapStatusToWarning(PatientStatus status)
+        {
+            return status switch
+            {
+                PatientStatus.NotFound => "Patient not found",
+                PatientStatus.Deceased => "Patient is deceased",
+                PatientStatus.NotUser => "Patient is not a user",
+                _ => null,
+            };
+        }
+
+        private void RetrievePatientData()
+        {
+            if (this.Patient?.Hdid != this.Hdid)
+            {
+                this.Dispatcher.Dispatch(new PatientSupportActions.ResetStateAction());
+                this.Dispatcher.Dispatch(new PatientSupportActions.LoadAction(PatientQueryType.Hdid, this.Hdid));
+            }
+
+            if (this.Patient?.Status == PatientStatus.Default)
+            {
+                this.Dispatcher.Dispatch(new MessageVerificationActions.ResetStateAction());
+                this.Dispatcher.Dispatch(new MessageVerificationActions.LoadAction(this.Hdid));
+            }
```

Comment:
I think these should be moved into the previous if block and the status check should be removed.

1. Only patients with a status of NotUser are guaranteed to have no UserProfiles.
2. If MessageVerificationActions.ResetStateAction is not called, you may end up with leftover messaging verifications still being displayed when querying a different patient
3. Due to the asynchronous nature of the calls, Patient may be null when this executes.
   - If you wanted or needed to issue the calls sequentially, you could subscribe to PatientSupportActions.LoadSuccessAction during this page's initialization, and have it call a method that dispatches MessageVerificationActions.LoadAction.

Reply:
Done